src/a11y-dialog.ts

```diff
@@ -3,22 +3,27 @@ import focusableSelectors from 'focusable-selectors'
 const TAB_KEY = 'Tab'
 const ESCAPE_KEY = 'Escape'
 
-/**
- * @typedef { import ('./a11y-dialog').A11yDialog } A11yDialogType
- * @typedef { import('./a11y-dialog').EventType } EventType
- * @typedef { import('./a11y-dialog').EventHandler } EventHandler
- * @typedef { import('./a11y-dialog').ListenersRecord } ListenersRecord
- */
+export type A11yDialogEvent = 'show' | 'hide' | 'destroy' | 'create'
+export type A11yDialogInstance = InstanceType<typeof A11yDialog>
+export type A11yDialogEventHandler = (node: Element, event?: Event) => void
+
+type ListenersRecord = Partial<
+  Record<A11yDialogEvent, A11yDialogEventHandler[]>
+>
 
-/** @type A11yDialogType */
 export default class A11yDialog {
-  /** @param {element} HTMLElement */
-  constructor(element) {
+  private $el: HTMLElement
```

Comment:
I don’t know if TypeScript will throw an error or if it will simply obfuscate the types of a private property – I think the latter. I can test that.

Reply:
Fair enough, thank you for the information. I’m wondering whether this is fine. I guess so?

Reply:
I’m fine with making `$el` public if you think authors will want access to it, but I think we need to document the risks of that , too. Alternatively: since we’re releasing a major version, now would be the time to privatize `$el` and tell authors they can’t rely on it.

Reply:
Yeah, I guess we might as well. We can always make it public later in an non-breaking way.

Reply:
Sure! For the record, I don’t want to come in here and boss my way around the architecture of your library, and I apologize if I have done that. This is your project and I will happily defer to you about what should be private and how things are named – and that includes the underscores, if you really like them 💜

Reply:
No, not at all! Genuinely happy to have someone else’s input on this, really. It’s interesting to have these discussions, and I’m not overly opinionated about it so really glad to get your input. 😊

Reply:
We could check [react-a11y-dialog](https://github.com/KittyGiraudel/react-a11y-dialog/), [svelte-a11y-dialog](https://github.com/AgnosticUI/svelte-a11y-dialog) and [vue-a11y-dialog](https://github.com/morkro/vue-a11y-dialog) to see if they rely on some of these newly private properties. Just to avoid making their work harder?

Reply:
Oh yes, can do! I was planning to get my gay little hands all over those wrappers anyway, as they need TypeScript love too 🔪

Reply:
Nice! I maintain the React port so probably a healthy place to start. @morkro maintains the Vue one with support from @roblevintennis I believe, who also maintains the Svelte one.

Reply:
Excellent. I am selfishly invested in the React port since I have a use case for it at work, but I’d love to work on the Vue port as well. I’m not experienced with Svelte, but I’ll pitch in there, too, if necessary ☺️